pkg/store/hintspb/custom.go

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package hintspb
+
+func (m *SeriesResponseHints) AddQueriedBlock(id string) {
```

Comment:
I would keep it typed and then decided how to format (.String()) below. It will make sure we will pass ULID nothing else (: